### PR TITLE
Fix user password storage, use OAuth tokens for GitHub

### PR DESCRIPTION
**Pull request by lgessler**
- Add passlib as a dependency
- Modify logintools to use passlib to hash passwords before writing to user files
- Generate GitHub OAuth token at new user creation time, store OAuth token and ID instead of GitHub password

When users upgrade and include this patch, their user configs' data model will have changed:

- `password` now stores the output of `pass_enc(custom_app_context.hash(password, salt=""))` instead of `pass_enc(password)`
- `git_password` is gone
- `git_token` and `git_id` are added

This data model change can be fixed with a short little script run at installation time, however. We could probably also detect whether we're using an old-style config in the code and then make the necessary changes on the fly. Let me know if you'd like to have either one of these before merging.

### CODE REVIEW

**Comment by amir-zeldes** [comment]
Thanks @lgessler - since we're working against dev here there's no problem doing incremental merges that are not 100% working, but for the sake of neatness / ease of figuring out what we did when+how, it might be best to do the backwards compatibility within this PR. 

I think whatever we can do automagically we should, since things will just work no questions asked. Auto checking a user's file when they login and updating as necessary shouldn't be hard. What I'm not 100% clear about is how the OAuth will be generated - are you suggesting auto inputting the GitHub credentials we have stored to generate the token, or prompting the user to authorize the token? We also need to consider the case when a user is upgraded from 'user' to 'committer' level, and how they authorize generating the token.

**Comment by lgessler** [comment]
OK, I'll add code to this PR to handle the data model change.

Re: generating the token, that's correct. All that is needed is a username and password to generate an OAuth token. No other user action is necessary. You can see how this happens in [the code changes to user creation](https://github.com/gucorpling/gitdox/blob/21c532a621d90eeafc6d9d140765612ffa59d637/admin.py#L67). You simply call `github3.authorize` with the user's credentials, which we can also do with no user intervention when we encounter an old-style config that has a GH username and password.

Re: the change from user to committer level, how exactly does that work? Does an admin recreate the account, or does the admin edit the user's ini file by hand and set `admin` to a value greater than `0`?

**Comment by lgessler** [comment]
Alright, so the commits got a bit messy there--I originally tried to make a standalone script before I thought of something better that would work automatically. Only pay attention to [these](https://github.com/gucorpling/gitdox/pull/79/commits/7ad4e5ec0e4552cd6991cf338677e084fdca57ae) [two](https://github.com/gucorpling/gitdox/pull/79/commits/b45ba764819f92b7984c047d2d89ac6d6e107b28) commits.

Basically, with these changes, when a user logs in, we check to see if they're using the old data model (fortunately there's enough information to tell), and if they are, we migrate them to the new one without any human intervention.

**Comment by amir-zeldes** [comment]
OK, I paused on 'fail silently', but then I thought, this is no different from what we have right now, so probably fine. If it's a connection issue I take it o.write() won't occur, so it will try again next time.

**Comment by amir-zeldes** [comment]
I forgot to say: at the moment changing the user/committer level from the admin interface does indeed amount to overwriting the user name, so I guess until we make that more elegant, this step doesn't require anything special. The tricky part is how to allow the newly minted committer use to enter their GitHub password: non-admin users can change their password in the 'reduced' admin form that's available to them, but I'm wondering if login will fail for them and they won't be able to reach that screen. Failing silently might mean that this is OK, but we should test.

**Comment by lgessler** [comment]
That is a problem! submitted a PR
